### PR TITLE
Perf: Improve iconIdForObject()...

### PR DESCRIPTION
**Pull request by krf**
Reduce allocations caused by QByteArray::fromRawData(...)

Task-Id: KDEND-171

### CODE REVIEW

File: core/util.cpp [hunk 1]
```diff
@@ -167,7 +169,9 @@ static IconDatabase readIconData()
         const QDir dir(fileInfo.path());
         const QString className = dir.dirName();
         const QString iconName = fileInfo.fileName();
-        IconCacheEntry &perClassData = data[className.toLatin1()];
+        const QByteArray classNameAsByteArray = className.toLatin1();
+        IconCacheEntry &perClassData = data[QLatin1String(classNameAsByteArray)];
```
Comment (marc-kdab) [inline core/util.cpp]: You might want to pass `data()` + `size()` here, as the `QLatin1String(QByteArray)` ctor performs a (useless) `strlen()`.

File: core/util.cpp [hunk 2]
```diff
@@ -167,7 +169,9 @@ static IconDatabase readIconData()
...
+        perClassData.className = classNameAsByteArray;
```
Comment (marc-kdab) [inline core/util.cpp]: std::move ?

File: core/util.cpp (hunk 1 above)
Comment (krf) [inline core/util.cpp]: Fixed.

File: core/util.cpp (hunk 2 above)
Comment (krf) [inline core/util.cpp]: Fixed.

**Pull request closed without merging**